debian/ubuntu-advantage-tools.postinst

```diff
@@ -354,6 +356,17 @@ remove_old_systemd_timers() {
         deb-systemd-helper purge ua-license-check.path > /dev/null || true
         deb-systemd-helper unmask ua-license-check.path > /dev/null || true
     fi
+
+    # If we're do-release-upgrad-ing to bionic, then clean up the xenial-only
+    # cloud-id-shim unit
+    if [ "$VERSION_ID" = "18.04" ]; then
+        if echo "$PREVIOUS_PKG_VER" | grep -q "16.04"; then
```

Comment:
This match against 16.04 seems fragile to me, but you can leave it as-is if you prefer.